src/JustSaying/Extensions/ChannelExtensions.cs

```diff
@@ -9,39 +10,49 @@ namespace JustSaying.Extensions
 {
     public static class ChannelExtensions
     {
-        public static void Merge<T>(
+        public static Task Merge<T>(
             IEnumerable<ChannelReader<T>> inputs,
             ChannelWriter<T> output,
             CancellationToken stoppingToken)
         {
             if (inputs is null) throw new ArgumentNullException(nameof(inputs));
             if (output is null) throw new ArgumentNullException(nameof(output));
-        
-            async IAsyncEnumerable<T> ReadAllAsync(ChannelReader<T> reader)
-            {
-                if (reader == null) throw new ArgumentNullException(nameof(reader));
-
-                while (await reader.WaitToReadAsync(stoppingToken).ConfigureAwait(false))
-                {
-                    while (reader.TryRead(out T item))
-                    {
-                        yield return item;
-                    }
-                }
-            }
 
-            Task.Run(async () =>
+            return Task.Run(async () =>
```

Comment:
I had a think about this, and I think returning the task doesn't make much sense here. There is a general rule of "don't do Task.Run, let your caller do it", and always return a task, but here the task is never intended to be awaited, it's more that this method kicks off the async pull and push mechanism, I think the usual rules do not apply.
It's a minor nit, I'm happy to leave it as it is.

Reply:
Happy to go either way here. I agree that this might be a bit more of a pragmatic call for encapsulation. Personally I think tasks should usually only be returned if they should be awaited, because the power of the message signature is to say that something is awaitable and returning it anyway is a bit confusing.